src/main/java/io/cryostat/agent/Registration.java

```diff
@@ -205,28 +194,17 @@ private DiscoveryNode defineSelf() throws UnknownHostException {
         return selfNode;
     }
 
-    @Override
-    public void stop() {
-        if (consumer != null) {
-            consumer.unregister();
-        }
-        log.info("{} stopped", getClass().getName());
-    }
+    void stop() {}
 
-    Future<Void> deregister() {
+    CompletableFuture<Void> deregister() {
         if (!this.pluginInfo.isInitialized()) {
-            return Future.succeededFuture();
+            log.info("Deregistration requested before registration complete!");
+            return CompletableFuture.completedFuture(null);
         }
         return cryostat.deregister(pluginInfo)
-                .onComplete(
-                        ar -> {
-                            if (this.webServerId != null) {
-                                getVertx().undeploy(this.webServerId);
-                            }
-                        })
-                .onComplete(
-                        ar -> {
-                            if (ar.failed()) {
+                .handleAsync(
+                        (n, t) -> {
+                            if (t != null) {
```

Comment:
Should we check if the return is null as well?

Reply:
The `n` value here is always null, actually. This is the only consumer and it doesn't need it, so the `CryostatClient` simply passes through `null` and `t` is all that's used to determine success vs failure.